### PR TITLE
allow person in contact list to be called by names instead of index in commands

### PR DESCRIPTION
**Pull request by elvern18**
#72

### CODE REVIEW

**Comment by elvern18** [comment]
completed for Delete, still working on Edit and View

File: src/main/java/seedu/address/logic/commands/DeleteByNameCommand.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredPersonList(predicate);
+        List<Person> updatedList = model.getFilteredPersonList();
+        if (updatedList.isEmpty()) {
+            return new CommandResult(String.format(Messages.MESSAGE_PERSON_NOT_FOUND));
+        }
+        if (updatedList.size() == 1) {
+            Person personToDelete = updatedList.get(0);
+            model.deletePerson(personToDelete);
```
Comment (joellow88) [inline src/main/java/seedu/address/logic/commands/DeleteByNameCommand.java:36]: Can consider adding `model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS)` here

**Review by joellow88 — COMMENTED** [review]
There is a bug where deleting a person by name causes all contacts to not be shown. Possible fix suggested in the code

File: src/main/java/seedu/address/logic/commands/DeleteByNameCommand.java (hunk 1 above)
Comment (elvern18) [reply src/main/java/seedu/address/logic/commands/DeleteByNameCommand.java:36]: Modified and tested.

**Review by Statspadders — APPROVED** [review]
LGTM and tested

**Pull request merged**